### PR TITLE
TBB Compatibility Update, master branch (2020.08.12.)

### PR DESCRIPTION
**Pull request by krasznaa**
Removed the forward declaration of [tbb::task_arena](https://software.intel.com/en-us/node/506359), as the "real class" is (a bit surprisingly) called `tbb::d1::task_arena` in [oneAPI](https://software.intel.com/oneapi)'s latest beta.

https://github.com/oneapi-src/oneTBB/blob/onetbb_2021/include/tbb/task_arena.h

In any case, hardcoding a name like `tbb::interface7::task_arena` just doesn't seem like a good idea...

### CODE REVIEW

**Comment by xvallspl** [comment]
Hi, Attila.

The forward declaration's purpose is to keep the tbb headers in the source files, so we would still need it. But I see the moment we were fearing, a change in the tbb::task_arena "real class", has arrived.

I will take a look at this, but just for my understanding, Is the oneTBB latest version a requirement for you?

**Comment by krasznaa** [comment]
Hi Xavier,

I would very much like to be able to use ROOT together with the latest oneAPI beta, if at all possible. Since oneAPI comes (amongst many other things) with TBB, if I want to use ROOT reliably in that environment, it should really use oneAPI's TBB version. (In the past this worked quite well for my tests.)

Wanting to hide the public TBB dependency from your interfaces is a good goal. But I would not do it with a forward declaration. Even in case the class would not be hidden behind version specific namespaces. If you do need to use the TBB class in the interface of the class, then tough luck. Then the class can just not hide its dependence on TBB, and that's the end of it. You could however try to design the class such that it would not have to directly reference the `tbb::task_arena` type at all in its interface. Then you could cleanly hide the include of this type into the source file.

Long story short: TBB is changing its interface, as we've been warned for a while. If ROOT wants to be able to use multiple versions of TBB at the same time, it must stop relying on explicit names like `tbb::interface7::task_arena`...

Cheers,
       Attila

**Comment by xvallspl** [comment]
Hi, Attila.

We discussed this for a while ourselves, and we designed the classes in the way it was more convenient for ROOT, willing to play the price of adapting to a change of interfaces in tbb when it happens. Of course, we might be wrong and we will review our choices :)

**Comment by hageboeck** [comment]
Converted to draft in order to be able to remove the `do not merge` label.

**Comment by xvallspl** [comment]
Addressed in https://github.com/root-project/root/pull/7062
